pkg/net/neighborhood.go

```diff
@@ -352,3 +340,26 @@ func (req leaseRequest) fail() {
 	close(req.done)
 	close(req.edge.cq)
 }
+
+func arePeersNear(id1 peer.ID, id2 peer.ID) bool {
+	id1Bytes, _ := id1.MarshalBinary()
+	id2Bytes, _ := id2.MarshalBinary()
+	xorId := xorShortestBytes(id1Bytes, id2Bytes)
+	return (xorId[0] & 1) == 1
```

Comment:
In these lines we simply get the bytes of the addresses and xor them. Just in case the address is malformed, we simply xor the amount of bytes of the shortest of the two addresses. Then, we don't compare whether the first byte is `1111 1111`, but we compare only whether the first bit is `1`. This gives a 50% probability instead of 1/256 (if we compared with `1111 1111`).

However I forgot we were treating with multihashes, so I think I should compare the last bit instead of the first one. Good point! 

Lastly, I do prefer comparing the last bit, instead of doing `<`, because in the future we may prefer not to use 50% probability. And for that, bits comparison is very easily extensible. For example, if you wanted 25% probability for using `head` selection strategy, you would simply take the first two bits and compare whether it is `0`.

Reply:
>Then, we don't compare whether the first byte is 1111 1111, but we compare only whether the first bit is 1.

Ah right ... duh 🤦‍♂️ 😄 

>However I forgot we were treating with multihashes, so I think I should compare the last bit instead of the first one. Good point!

👍 

>Lastly, I do prefer comparing the last bit, instead of doing <, because in the future we may prefer not to use 50% probability. And for that, bits comparison is very easily extensible.

Upon reflection, I think you're right.   Flexibility is key at this point.  👍